stackslib/src/chainstate/stacks/transaction.rs

```diff
@@ -707,19 +710,24 @@ impl StacksTransaction {
                 )));
             }
         };
+        let tx = StacksTransaction {
+            version,
+            chain_id,
+            auth,
+            anchor_mode,
+            post_condition_mode,
+            post_conditions,
+            payload,
+        };
 
-        Ok((
-            StacksTransaction {
-                version,
-                chain_id,
-                auth,
-                anchor_mode,
-                post_condition_mode,
-                post_conditions,
-                payload,
-            },
-            fd.num_read(),
-        ))
+        if !StacksBlock::validate_transactions_static_epoch(&vec![tx.clone()], epoch_id, false) {
```

Comment:
It's not necessary to use a `Vec` here, use a slice and avoid the heap allocation:

```suggestion
        if !StacksBlock::validate_transactions_static_epoch(&[tx.clone()], epoch_id, false) {
```

This applies to other uses of this function as well